app/services/setting_registry.rb

```diff
@@ -104,12 +104,15 @@ def load_values
       definition.updated_at = s.updated_at
       definition.value = s.value
     end
-    # load nil to set value to default
-    @settings.except(*loaded_names).each do |name, definition|
-      # Creating missing records as we operate over the DB model while updating the setting
-      _new_db_record(definition).save(validate: false)
-      definition.updated_at = nil
-      definition.value = definition.default
+
+    if create_records
+      # load nil to set value to default
+      @settings.except(*loaded_names).each do |name, definition|
+        # Creating missing records as we operate over the DB model while updating the setting
+        _new_db_record(definition).save(validate: false)
+        definition.updated_at = nil
+        definition.value = definition.default
+      end
```

Comment:
How about instead of adding a parameter, move this to a separate function with
```suggestion
      @settings.except(Setting.unscoped.all.pluck(:name)).each do |name, definition|
        # Creating missing records as we operate over the DB model while updating the setting
        _new_db_record(definition).save(validate: false)
        definition.updated_at = nil
        definition.value = definition.default
      end
```

Reply:
Nice, make sense and allows us to get it off of this function :+1: